add restaurant filter button

rianp opened the pull request:
## Changes made:

1. Added restaurant filter button.

## Comments:
When you click the Restaurant Filter button, it toggles open to reveal the text "filters." The button utilizes a useState hook and a function called toggleFilterCard. Also, I've created a simple test for this button that you can execute.

rianp committed:
add restaurant filter button

clairenguyen commented on frontend/__tests__/components/CardScreen.test.tsx:
```diff
@@ -51,4 +51,12 @@ describe('CardScreen', () => {
+  test('clicking on restaurant filter button opens filter options', async () => {
+    expect(screen.queryByText('Filters')).not.toBeInTheDocument()
+
+    await userEvent.click(screen.getByRole('button', { name: 'Filter Restaurants' }))
```
I'm getting a `Unresolved variable or type userEvent` error. `userEvent` needs to be imported.  Were you able to run tests without it being imported?

rianp replied:
oh! totally! thanks for letting me know!

clairenguyen reviewed (changes requested):
The `CardScreen.test.tsx` and `__tests__/page.test.tsx`  are failing on my end. I'm assuming it might have passed before your merge from the main branch, but now is failing after you merged?  Could you try to resolve that?

rianp committed:
refactor test

clairenguyen commented:
@rianp  when you run `npm test` in the frontend, do all your tests pass? i'm still failing on `__tests__/page.test.tsx`

rianp commented:
hmm @clairenguyen they're all passing on my end. I'll look into this more. thanks for pointing this out

rianp commented:
@clairenguyen okay I'm pretty sure i fixed the issue!🤞

clairenguyen commented:
@rianp it works on my side now!  what did you end up doing?  I don't see any significant changes in the diff.

seoulection commented:
@rianp I noticed the changes to the `package-lock.json` file. Could you please verify if your local version of Node is `20.4.0`?

rianp committed:
update package-lock

rianp commented:
@clairenguyen I renamed the button to fix the test failures. It displayed "restaurants" when the filter switch was toggled in other tests. To resolve this, I changed the reveal text to "filters," as this information is temporary until menu options are added anyway.

clairenguyen commented:
@rianp awesome, thanks! looks good.
